packages/mattermost-redux/src/selectors/entities/threads.ts

```diff
@@ -66,16 +67,24 @@ export function getThreads(state: GlobalState): IDMappedObjects<UserThread> {
     return state.entities.threads.threads;
 }
 
-export function getThread(state: GlobalState, threadId: $ID<UserThread> | undefined): UserThread | null {
-    if (
-        !threadId ||
-        !(getThreadsInCurrentTeam(state)?.includes(threadId) || getUnreadThreadsInCurrentTeam(state)?.includes(threadId))
-    ) {
-        return null;
-    }
+export const getThread: (state: GlobalState, threadId: $ID<UserThread> | undefined) => UserThread | null = createSelector(
```

Comment:
Since we're memoizing this now, `getThread` should be turned into a selector factory called `makeGetThread`.

Alternatively, we'd probably be fine to just remove the `createSelector` entirely since none of this computation is particularly complex, and it's not returning a newly created object.

Reply:
you're right, removed memoization to minimize the changes.